Mathlib/Data/Fin/Tuple/Basic.lean

```diff
@@ -893,6 +951,12 @@ theorem preimage_insertNth_Icc_of_not_mem {i : Fin (n + 1)} {x : α i} {q₁ q
     simp only [mem_preimage, insertNth_mem_Icc, hx, false_and_iff, mem_empty_iff_false]
 #align fin.preimage_insert_nth_Icc_of_not_mem Fin.preimage_insertNth_Icc_of_not_mem
 
+theorem insertNth_comp_rev {α} (i : Fin (n + 1)) (x : α) (p : Fin n → α) :
+    (Fin.insertNth i x p) ∘ Fin.rev = Fin.insertNth (Fin.rev i) x (p ∘ Fin.rev) := by
+  funext x
+  simp
+  rw[insertNth_rev]
```

Comment:
Non-terminal `simp`s without `only` are not allowed in Mathlib. Please try one of these suggestions:
```suggestion
  apply insertNth_rev
```
```suggestion
  simp [insertNth_rev]
```
or even use `funext <| insertNth_rev ..` as a term-mode proof.
Also, please move this lemma right after `insertNth_rev`. Otherwise LGTM. Thanks!

bors d+